models/status.go

```diff
@@ -157,6 +158,77 @@ func GetLatestCommitStatus(repo *Repository, sha string, page int) ([]*CommitSta
 	return statuses, x.In("id", ids).Find(&statuses)
 }
 
+// GetIssuesLatestCommitStatuses returns all statuses with given repoIDs and shas
+func GetIssuesLatestCommitStatuses(issues []*Issue) ([][]*CommitStatus, error) {
```

Comment:
Why not use `GetLatestCommitStatus` ?

If you need to fetch from multiple SHAs at the same time, just make a `GetLatestCommitsStatus(repo *Repository, shas []string, page int)` and fetch the `shas` before that 🤔

Reply:
Also, I would return a `map[string][]*CommitStatus` to that you can `ret[shas[0]` when using them.

Reply:
OR, you can have a `func (*Issue) PopulateLatestCommitStatus()` that fetches them per Issue 🤔